client/src/components/TrackGroup/PurchaseOrDownloadAlbumModal.tsx

```diff
@@ -22,7 +23,7 @@ const PurchaseOrDownloadAlbum: React.FC<{
   const { state: artistState } = useArtistContext();
 
   const userId = user?.id;
-
+  const { trackGroupId } = useParams();
```

Comment:
Why not use `const trackGroupId = trackGroup.id`?

Reply:
Don't know... I did this cause it worked.

Reply:
`useParams` means it would only get a trackGroupId if it was defined in the URL. I'd use `trackGroup.id` and add a variable to the component (as described below)